central/reports/validation/validate.go

```diff
@@ -79,10 +84,14 @@ func ValidateCancelReportRequest(reportSnapshotStore reportSnapshotDS.DataStore,
 	return nil
 }
 
-func generateReportSnapshot(config *storage.ReportConfiguration, requester *storage.SlimUser,
-	collection *storage.ResourceCollection, protoNotifiers []*storage.Notifier,
+func generateReportSnapshot(
+	config *storage.ReportConfiguration,
+	requester *storage.SlimUser,
+	collection *storage.ResourceCollection,
+	protoNotifiers []*storage.Notifier,
 	notificationMethod storage.ReportStatus_NotificationMethod,
-	requestType storage.ReportStatus_RunMethod) *storage.ReportSnapshot {
+	requestType storage.ReportStatus_RunMethod,
+) *storage.ReportSnapshot {
```

Comment:
In ReportSnapshot to, we have the `notifier_id` stored in `EmailNotifierConfiguration`. You would need to change lines [ln120-122](https://github.com/stackrox/stackrox/pull/7370/files#diff-dea42debb1153658657cda8f7237e80ad1a2fed701f40b55079ba889dc93eb2dR120-R122) to assign `NotifierSnapshot.EmailConfig.NotifierId = protoNotifier.GetId()`